src/soundio/soundmanagerutil.h

```diff
@@ -102,27 +103,52 @@ class AudioPath {
         return qHash(path.hashValue(), seed);
     }
 
-    friend bool operator==(
-            const AudioPath& lhs,
-            const AudioPath& rhs) {
-        // Exclude m_channelGroup from comparison!
-        // See also: hashValue()/qHash()
-        // TODO: Why??
-        return lhs.m_type == rhs.m_type &&
-                lhs.m_index == rhs.m_index;
-    }
+    // CppCoreGuidelines C.161: Use non-member functions for symmetric operators
+    friend constexpr std::weak_ordering operator<=>(const AudioPath& lhs,
+            const AudioPath& rhs) noexcept;
+    friend constexpr bool operator==(const AudioPath& lhs,
+            const AudioPath& rhs) noexcept;
 
-protected:
+  protected:
     virtual void setType(AudioPathType type) = 0;
     ChannelGroup m_channelGroup;
     AudioPathType m_type;
     unsigned char m_index;
 };
 
-inline bool operator!=(
-        const AudioPath& lhs,
-        const AudioPath& rhs) {
-    return !(lhs == rhs);
+constexpr std::weak_ordering operator<=>(const AudioPath& lhs,
+        const AudioPath& rhs) noexcept {
+    // Exclude m_channelGroup from comparison!
+    // See also: hashValue()/qHash()
+    // TODO: Why??
+
+    // TODO: remove this ifdef when AppleClang ships a libc++
+    // implementation with operator<=> for tuples (XCode 14 probably)
+#ifdef __APPLE__
```

Comment:
Right, but its bad practice to only define the ordering partially and `operator<=>` is better than having to define all six ordering overloads manually (especially considering the `friend` boilerplate).

Reply:
CI passes on all platforms, do you still insist on the manual implementation?

Reply:
It fails also on Ubuntu Focal unfortunately. It knows nothing about <=>

> Right, but its bad practice to only define the ordering partially

It is also bad practice to define unused functions. 

So in this case I would just replace `<=>` by `operator<` for all and nothing else. 
This is IMHO enough and does not clutters the code with #ifdefs for code that is not used anyway. 

I will do a PR, for this because you are not able to test it. 
You can than use it to remove the Apple conditions in a fixup.

Reply:
Here it is: https://github.com/Swiftb0y/mixxx/pull/17 tested and fully working.